### ISSUE TITLE
GLFWscrollfun and mods

### ISSUE THREAD

**Issue by dv343** [glfw/glfw#454]
Now, GLFWscrollfun only sends the scroll offsets. It would be very useful to also send the keyboard mods since it is very common to change the app behavior with this (for example: firefox, chrome...).

**Comment by elmindreda** [comment]
Yes, that was an oversight on my part. It will be fixed in 4.0.

**Comment by shurcooL** [comment]
Is it accurate to say that as a general rule, (in future API revisions) it makes sense to send state of mod keys with all input events?

**Comment by elmindreda** [comment]
@shurcooL As a general rule, for input events, I'd say yes. Path dropping is one exception. Tablet input may be another.

**Comment by shurcooL** [comment]
Thanks for insight.

I agree about path dropping, but I think mods may still be useful for tablet input. I can imagine a mode where holding alt while dragging on a tablet pans the canvas rather than drawing on it, etc.

**Comment by elmindreda** [comment]
I agree that modifier bits would be useful for tablet input, but if the bits are polled at event processing time instead of saved by the platform at event generation time, they may be wrong.

**Comment by SephiRok** [comment]
+1 on this. Mouse wheel can be given lots of additional functionality with modifier keys in games.

Related: Would it be useful to add a function to get the last reported state of mods -- like glfwGetKey is there for keys?

**Comment by ws909** [comment]
> Path dropping is one exception 

Not really. I can imagine holding down a modifier key to change how the content is pasted. Simply dropping something into the window makes the application paste the file contents (image, for instance) into its view, while holding down option/alt/shift pastes text content instead, such as the file path.

**Comment by elmindreda** [comment]
That's true.

**Comment by ws909** [comment]
Ah. I stated the obvious. 
>  I can imagine a setup where holding alt while dragging on a tablet pans the canvas rather than drawing on it, etc.
It was already mentioned. Wow, sorry for the disturbance. I am really blind.
